modules/contact/test/tests/bouncing-block-contact/frictionless-weighted-gap-mixed-basis.i

```diff
@@ -0,0 +1,173 @@
+starting_point = 2e-1
+offset = 1e-2
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+  diffusivity = 1e0
+  scaling = 1e0
+[]
+
+[Mesh]
+  file = long-bottom-block-1elem-blocks.e
+  second_order = true
+  patch_update_strategy = always
+[]
+
+[Variables]
+  [./disp_x]
+    block = '1 2'
+    order = SECOND
+  [../]
+  [./disp_y]
+    block = '1 2'
+    order = SECOND
+  [../]
+  [./normal_lm]
+    block = 3
+  [../]
+[]
+
+[ICs]
+  [./disp_y]
+    block = 2
+    variable = disp_y
+    value = ${fparse starting_point + offset}
+    type = ConstantIC
+  [../]
+[]
+
+[Kernels]
+  [./disp_x]
+    type = MatDiffusion
+    variable = disp_x
+  [../]
+  [./disp_y]
+    type = MatDiffusion
+    variable = disp_y
+  [../]
+[]
+
+
+[Constraints]
+  [./weighted_gap_lm]
+    type = ComputeWeightedGapLMMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_x
+    secondary_disp_y = disp_y
+    use_displaced_mesh = true
+  [../]
+  [./ncp_lm]
+    type = ApplyPenetrationConstraintLMMechanicalContact
+    secondary = 10
+    primary = 20
+    variable = normal_lm
+    primary_variable = disp_x
+  [../]
+  [normal_x]
+    type = NormalMortarMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+  []
+  [normal_y]
```

Comment:
I am trying to understand why we still need `normal_x` and `normal_y`.  I feel `weighted_gap_lm` and `ncp_lm` are enough

Reply:
The `ComputeWeightedGapLMMechanicalContact` and `ApplyPenetrationConstraintLMMechanicalContact` objects together determine the value of the LM, e.g. they compute the LM residual. However, they do not apply the LM (which represents the contact pressure) to the displacement residuals/equations. That is what `normal_x` and `normal_y` do. They apply the contact force/pressure represented by the LM to the displacement equations.